fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/serializer/JsonElectionSerializer.java

```diff
@@ -0,0 +1,152 @@
+package com.github.dedis.popstellar.model.network.serializer;
+
+import com.github.dedis.popstellar.model.network.method.message.data.election.*;
+import com.github.dedis.popstellar.model.objects.Channel;
+import com.github.dedis.popstellar.model.objects.Election;
+import com.github.dedis.popstellar.model.objects.event.EventState;
+import com.github.dedis.popstellar.model.objects.security.MessageID;
+import com.github.dedis.popstellar.model.objects.security.PublicKey;
+import com.google.gson.*;
+
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class JsonElectionSerializer
+    implements JsonSerializer<Election>, JsonDeserializer<Election> {
+
+  @Override
+  public Election deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+      throws JsonParseException {
+    JsonObject jsonObject = json.getAsJsonObject();
+
+    // Deserialize the nested Channel object
+    Channel channel = context.deserialize(jsonObject.get("channel"), Channel.class);
+
+    // Deserialize the primitive values
+    String id = jsonObject.get("id").getAsString();
+    String name = jsonObject.get("name").getAsString();
+    long creation = jsonObject.get("creation").getAsLong();
+    long start = jsonObject.get("start").getAsLong();
+    long end = jsonObject.get("end").getAsLong();
+    JsonElement electionKeyElement = jsonObject.get("electionKey");
+    String electionKey = electionKeyElement == null ? null : electionKeyElement.getAsString();
+
+    // Deserialize the enums
+    ElectionVersion electionVersion =
+        ElectionVersion.valueOf(jsonObject.get("electionVersion").getAsString());
+    String stateString = jsonObject.get("state").getAsString();
+    EventState state = stateString.isEmpty() ? null : EventState.valueOf(stateString);
+
+    // Deserialize collections
+    JsonArray electionQuestionsJsonArray = jsonObject.get("electionQuestions").getAsJsonArray();
+    List<ElectionQuestion> electionQuestions = new ArrayList<>();
+    for (JsonElement electionQuestionJsonElement : electionQuestionsJsonArray) {
+      electionQuestions.add(
+          context.deserialize(electionQuestionJsonElement, ElectionQuestion.class));
+    }
+
```

Comment:
You could add a comment about what is being deserialized in the codeblocks below as you did for the previous ones.